src/main/java/fr/lunatech/timekeeper/services/EventTemplateService.java

```diff
@@ -64,38 +66,101 @@
+        final Optional<EventTemplate> maybeEvent = findEventById(eventId, ctx);
+
+        return maybeEvent.map(event -> {
+            boolean checkEvent = validateEventUpdation(
+                    newName,
+                    event.id,
+                    newStartTime,
+                    newEndTime,
+                    ctx);
+
+            if (checkEvent) {
+                throw new UpdateResourceException(
+                        "Event with name: " + request.getName() +
+                        ", already exists with same Start: " + request.getStartDateTime().toLocalDate() +
+                        "and End: " + request.getEndDateTime().toLocalDate() + " dates."
+                );
+            }
+            // delete all userEvent associated to the previous state of this event template
+            deleteAttendees(event);
+
+            // actually update the event. by side effect every userEvent associated will be created
+            EventTemplate eventTemplateUpdated = request.unbind(event, userService::findUserById, ctx);
+            eventTemplateUpdated.persistAndFlush();
+
+            return eventTemplateUpdated.id;
+        });
+    }
 
-        // actually update the event. by side effect every userEvent associated will be created
-        EventTemplate eventTemplateUpdated = request.unbind(maybeEvent.get(),userService::findById, ctx);
-        eventTemplateUpdated.persistAndFlush();
 
-        return Optional.of(eventTemplateUpdated.id);
+    private boolean validateEventUpdation(String name, Long eventId, LocalDateTime startDateTime, LocalDateTime endDateTime, AuthenticationContext ctx) {
+        final var startDate = startDateTime.toLocalDate();
+        final var endDate = endDateTime.toLocalDate();
+        final var foundEvents = findAllEventsByName(name, ctx)
+                .stream()
+                .filter(eventTemplate -> !eventTemplate.id.equals(eventId) &&
+                        eventTemplate.startDateTime.toLocalDate().isEqual(startDate) &&
+                        eventTemplate.endDateTime.toLocalDate().isEqual(endDate))
+                .findAny();
+
+        return foundEvents.isPresent();
+    }
+
+    private boolean validateEventCreation(String name, LocalDateTime startDateTime, LocalDateTime endDateTime, AuthenticationContext ctx) {
+        final var startDate = startDateTime.toLocalDate();
+        final var endDate = endDateTime.toLocalDate();
+
+        final var foundEvents = findAllEventsByName(name, ctx)
+                .stream()
+                .filter(eventTemplate -> startDate.isEqual(eventTemplate.startDateTime.toLocalDate()) &&
+                        endDate.isEqual(eventTemplate.endDateTime.toLocalDate()))
+                .findAny();
+
+        return foundEvents.isPresent();
     }
 
-    private Optional<EventTemplate> findById(Long id, AuthenticationContext ctx) {
+    private Optional<EventTemplate> findEventById(Long id, AuthenticationContext ctx) {
```

Comment:
findById was correct because you are using the service that manipulates the Events. Also it is private. It should be findById according to our convention.